### PR TITLE
feat(core): add i18n foundation with English locale and UI wiring

### LINKED ISSUE #1363: Feature: Core i18n Types, Translator, and Registry
Foundation for Video.js 10 internationalization. Adds the core types, pure translator function, and global registry that platform packages (`@videojs/html`, `@videojs/react`) and components build on.

### Types (`@videojs/core/i18n/types.ts`)

- `BuiltInLocale` — union of shipped locales (`'ar' | 'de' | 'es' | 'fr' | 'it' | 'ja' | 'ko' | 'nl' | 'pl' | 'pt' | 'ru' | 'tr' | 'zh'`)
- `Locale` — `BuiltInLocale | (string & {})` for BCP 47 autocomplete
- `Translations` — all standard player keys, all optional; value types use `Contains<>` helper to enforce `{param}` placeholders
- `TranslationParams` — per-key parameter contract (key → required params, or `never`)
- `Translator` — callable type with overloaded signatures so keys with params require the exact params object
- `Contains<Needle>` template-literal helper for substring constraints

### Translator (`@videojs/core/i18n/translator.ts`)

- `createTranslator(translations, locale)` — pure function, runtime `{param}` interpolation via `String.prototype.replace`
- Type-only overloads for the `Translator` shape; loose implementation cast at the return site

### Registry (`@videojs/core/i18n/registry.ts`)

- Singleton `Map<string, Partial<Translations>>`
- `registerI18n(locale, translations)` — merging (not replacing) writes
- `getI18nTranslations(locale)` — BCP 47 left-truncation fallback (`es-419-u-nu-latn → es-419 → es → en`)
- `onI18nRegistryChange(cb)` — subscription with unsubscribe function
- `hasRegisteredI18n(locale)` — predicate

### English defaults (`@videojs/core/i18n/locales/en.ts`)

- All keys from `Translations` populated with English strings
- Pre-registered at module import of `@videojs/core/i18n/registry`

## Tests

- Unit tests for `createTranslator` — key lookup, fallback to key, `{param}` substitution
- Unit tests for registry — register/retrieve, merging, subscription, `hasRegisteredI18n`
- BCP 47 fallback edge cases — `zh-Hant-HK`, `en-GB-scotland`, `es-419`, sibling non-fallback (`es-AR` vs registered `es-MX`)

## References

- Design doc: `internal/design/i18n.md`

### PR DESCRIPTION
**Pull request by sampotts**
Refs #222
Closes #1363

## Summary

Foundation PR for Video.js 10 i18n: registry, translator, English copy, browser Translation API fallback, locale-aware formatting utilities, and opaque translation keys wired through core UI components.

## Changes

- `@videojs/core/i18n` — registry, `createTranslator`, BCP 47 lookup chain, `en.ts` base layer
- Browser Translation API fallback with placeholder masking and cache
- `@videojs/utils` — DOM locale helpers (`subscribeAmbientLang`, `mergeLocaleOverlays`) and `formatDuration` via `Intl.DurationFormat`
- Core UI controls use translation keys for aria labels, tooltips, error dialog copy, and input-feedback indicators
- Design doc updated to `implemented` status

<details>
<summary>Stack info</summary>

This is **PR 1 of 5** in the i18n stack. Merge order:

1. **Core** (this PR) → 2. Locale packs → 3. HTML → 4. React → 5. Sandbox

</details>

## Testing

```bash
pnpm -F @videojs/core build
pnpm -F @videojs/core test src/core/i18n src/core/ui
pnpm -F @videojs/utils test src/dom src/time
pnpm typecheck
```

Made with [Cursor](https://cursor.com)

---

> [!NOTE]
> **Medium Risk**
> Touches accessibility strings and media error messaging across many UI cores; behavior depends on downstream HTML/React translators, but English defaults and tests limit user-visible regressions in this PR alone.
> 
> **Overview**
> Adds **`@videojs/core/i18n`**: a global registry with BCP 47 lookup, typed **`createTranslator`**, English defaults in **`en.ts`**, and optional **Browser Translation API** fallback (placeholder masking, `shouldAttemptBrowserTranslation`, pre-installed models only).
> 
> **Core UI** now exposes **opaque translation keys** (and param hooks like `getLabelParams` / `getValueTextParams`) instead of hardcoded English for play/mute/seek/sliders/time, live/cast/captions, playback rate, error dialog copy, and input-feedback indicators. **`resolveControlAttrs`** / **`resolveOptionalControlLabel`** bridge keys to resolved strings for platform layers.
> 
> **`@videojs/utils`** gains locale DOM helpers (`effectiveLocale`, `mergeLocaleOverlays`, ambient `lang` subscription) and **`formatDuration`** / **`formatVolumePercent`** via **`Intl`**. Native HLS errors for standard codes are normalized to canonical messages so dialog copy can map to registry keys.
> 
> **`internal/design/i18n.md`** is marked **implemented** and documents keys, merge order, and tooltip/`media-text` behavior. **`@videojs/core`** exports **`./i18n`** and **`./i18n/locales/*`**; registry is a declared **side effect**. Non-English locale packs and HTML/React providers are explicitly deferred to follow-up stack PRs.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b58714d5bdd814a55051cf637a943d6ff4a3cc9a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/core/src/core/i18n/browser-translation.ts
Comment (sampotts) [inline packages/core/src/core/i18n/browser-translation.ts:1]: Happy to remove this if we don't think it's worthwhile. It's a bit of an edge case solution anyway. For it to work currently:
- the user has to be using Chrome; and
- they have to already have the language pack downloaded (we won't trigger it)

File: packages/core/tsdown.config.ts [hunk 1]
```diff
@@ -3,10 +3,16 @@ import { defineConfig } from 'tsdown';
 import { type PackageBuildMode, packageBuildConfig, packageBuildModes } from '../../build/tsdown.ts';
 import packageJson from './package.json' with { type: 'json' };
 
+const localeEntries = {
+  'i18n/locales/en': './src/core/i18n/locales/en.ts',
```
Comment (sampotts) [inline packages/core/tsdown.config.ts:7]: More are added in later PRs 😅

File: packages/html/src/ui/time-slider/tests/time-slider-element.test.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ describe('TimeSliderElement', () => {
 
   it('initializes with default property values', () => {
     const slider = createElement(TimeSliderElement);
-    expect(slider.label).toBe('Seek');
+    expect(slider.label).toBe('');
```
Comment (sampotts) [inline packages/html/src/ui/time-slider/tests/time-slider-element.test.ts:30]: Empty string means it'll be translated with the default key.

File: packages/core/src/core/i18n/browser-translation.ts
Comment (luwes) [reply packages/core/src/core/i18n/browser-translation.ts:1]: I'm not sure, context matters sometimes with translations. That browser API might give some strange results?

File: packages/core/src/core/i18n/registry.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+/** Registry map key: normalized tag with unicode extensions removed (same base as {@link localeLookupChain}). */
+function canonicalLocaleRegistryKey(locale: string): string {
+  return stripUnicodeExtensions(normalizeLocaleTag(locale));
+}
+
+/**
+ * Most-specific-first BCP 47 lookup tags (normalized). Always ends with `en` when missing from the truncated chain.
+ *
+ * @example `es-419-u-nu-latn` → `['es-419', 'es', 'en']`
```
Comment (luwes) [inline packages/core/src/core/i18n/registry.ts:36]: 👍

File: packages/core/src/core/i18n/locales/en.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+/** Default English layer — registered when `@videojs/core/i18n/registry` (or this package entry) is loaded. */
+export default {
+  play: 'Play',
+  pause: 'Pause',
+  replay: 'Replay',
+  mute: 'Mute',
+  unmute: 'Unmute',
+  seekForward: 'Seek forward {seconds} seconds',
+  seekBackward: 'Seek backward {seconds} seconds',
+  enterFullscreen: 'Enter fullscreen',
```
Comment (luwes) [inline packages/core/src/core/i18n/locales/en.ts:12]: (blocking) I didn't expect this type of keys. Why not use both `Enter fullscreen` for key and value?
this will be a lot easier for translations because the key is the english value to translate.
in code this will read well too, maybe long in some cases.
(feedback is about all keys in this file)

Comment (sampotts) [reply packages/core/src/core/i18n/locales/en.ts:12]: We came to the decision to use opaque keys rather than English:
https://github.com/videojs/v10/blob/main/internal/design/i18n.md#opaque-camelcase-keys-not-english-strings

File: packages/core/src/core/i18n/browser-translation.ts
Comment (sampotts) [reply packages/core/src/core/i18n/browser-translation.ts:1]: Yeah, that's valid. What I've done in the past is have a "Description" field alongside the key to set the context as translators often ask how it's being used (e.g. verb vs noun). Maybe we could add that somehow.

File: packages/html/src/ui/playback-rate-radio-group/playback-rate-radio-group-element.ts [hunk 5]
```diff
@@ -54,8 +54,11 @@ export class PlaybackRateRadioGroupElement extends MenuRadioGroupElement {
 
       this.value = this.#core.getRateValue(state.rate);
       if (!this.hasAttribute('aria-label') && !this.hasAttribute('aria-labelledby')) {
-        this.setAttribute('aria-label', 'Playback rate');
+        this.setAttribute('aria-label', this.#core.getLabel(state));
```
[the remark this answers is not in the document]
Comment (sampotts) [reply packages/html/src/ui/playback-rate-radio-group/playback-rate-radio-group-element.ts:57]: Yep, we have to merge everything at once so once all the stacked PRs are approved I'll squash them into this one and then merge.

**Review by luwes — APPROVED** [review]
LGTM thanks @sampotts !
